code/datums/chatmessage.dm

```diff
@@ -254,19 +262,34 @@ GLOBAL_LIST_INIT(job_colors_pastel, list(
 	spans = spans ? spans.Copy() : list()
 
 	// Check for virtual speakers (aka hearing a message through a radio)
-	var/atom/movable/originalSpeaker = speaker
 	if (istype(speaker, /atom/movable/virtualspeaker))
 		var/atom/movable/virtualspeaker/v = speaker
+		//===================
+		//Check to make sure we didnt hear the source message
+		//===================
+		//Dont create the overhead chat if we said the message.
+		if(v.source == src)
+			return
+		//Dont create the overhead radio chat if we are a ghost and can hear global messages.
+		if(isobserver(src) && !(client.prefs.chat_toggles & CHAT_GHOSTEARS))
+			return
+		//Dont create the overhead radio chat if we heard the speaker speak
+		if(get_dist(get_turf(v.source), get_turf(src)) <= 1)
+			return
```

Comment:
This will confuse people. I don't think seeing radio and whisper at the same time is such a big problem.

Reply:
This doesnt make it show up as a whisper, it makes the original show up as a radio message. Just without the poor thought around double messages.